albumentations/augmentations/transforms.py

```diff
@@ -3177,3 +3178,133 @@ def get_transform_init_args_names(self):
+        shift_y: int = 0,
+        random_offset: bool = False,
+        fill_value: int = 0,
+        always_apply: bool = False,
+        p: float = 0.5,
+    ):
+        super(GridDropout, self).__init__(always_apply, p)
+        self.ratio = ratio
+        self.unit_size_min = unit_size_min
+        self.unit_size_max = unit_size_max
+        self.holes_number_x = holes_number_x
+        self.holes_number_y = holes_number_y
+        self.shift_x = shift_x
+        self.shift_y = shift_y
+        self.random_offset = random_offset
+        self.fill_value = fill_value
+        assert 0 < self.ratio < 1, "ratio must be between 0 and 1."
+
+    def apply(self, image, fill_value=0, holes=[], **params):
+        return F.cutout(image, holes, fill_value)
+
+    def get_params_dependent_on_targets(self, params):
+        img = params["image"]
+        height, width = img.shape[:2]
+        # set grid using unit size limits
+        if self.unit_size_min and self.unit_size_max:
+            assert (
+                2 <= self.unit_size_min <= self.unit_size_max
+            ), "Max unit size should be >= min size, both at least 2 pixels."
+            assert self.unit_size_max <= min(height, width), "Grid size limits must be within the shortest image edge."
+            self.unit_width = random.randint(self.unit_size_min, self.unit_size_max + 1)
+            self.unit_height = self.unit_width
+        else:
+            # set grid using holes numbers
+            if self.holes_number_x is None:
+                self.unit_width = max(2, width // 10)
+            else:
+                assert (
+                    1 <= self.holes_number_x <= width // 2
+                ), "The hole_number_x must be between 1 and image width//2."
+                self.unit_width = width // self.holes_number_x
+            if self.holes_number_y is None:
+                self.unit_height = max(min(self.unit_width, height), 2)
+            else:
+                assert (
+                    1 <= self.holes_number_y <= height // 2
+                ), "The hole_number_y must be between 1 and image height//2."
+                self.unit_height = height // self.holes_number_y
+
+        self.hole_width = int(self.unit_width * self.ratio)
+        self.hole_height = int(self.unit_height * self.ratio)
+        # min 1 pixel and max unit length - 1
+        self.hole_width = min(max(self.hole_width, 1), self.unit_width - 1)
+        self.hole_height = min(max(self.hole_height, 1), self.unit_height - 1)
+        # set offset of the grid
+        if self.shift_x is None:
+            self.shift_x = 0
+        if self.shift_y is None:
+            self.shift_y = 0
+        if self.random_offset:
+            self.shift_x = random.randint(0, self.unit_width)
+            self.shift_y = random.randint(0, self.unit_height)
+        else:
+            self.shift_x = min(max(0, self.shift_x), self.unit_width)
+            self.shift_y = min(max(0, self.shift_y), self.unit_height)
+
```

Comment:
Here you rewrite values, thats was defined by user. Remove `self` qualifier for new `shift_x` and `shift_y`